src/v/kafka/server/handlers/handler_probe.cc

```diff
@@ -111,6 +117,32 @@ void handler_probe::setup_metrics(
       });
 }
 
+// For public metrics we only expose a small subset of the metrics for produce
+// and fetch requests to keep the count of metrics low
+void handler_probe::setup_public_metrics(
+  ssx::metrics::metric_groups& metrics, api_key key) {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
```

Comment:
Shouldn't this be `disable_public_metrics()`?

Reply:
Yes, forgot there is two different flags. Fixed.

Reply:
If everyone didn't have to remember this whole song and dance but could just do `rp_metrics::add_histo("kafka_handler", _latency, INCLUDE_IN_PUBLIC)` it would be even better, but I've beaten that drum enough :).